cmd/zetacored/parse_genesis.go

```diff
@@ -0,0 +1,189 @@
+	slashingtypes "github.com/cosmos/cosmos-sdk/x/slashing/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
+	evmtypes "github.com/evmos/ethermint/x/evm/types"
+	feemarkettypes "github.com/evmos/ethermint/x/feemarket/types"
+	"github.com/spf13/cobra"
+	"github.com/tendermint/tendermint/types"
+	"github.com/zeta-chain/zetacore/app"
+	crosschaintypes "github.com/zeta-chain/zetacore/x/crosschain/types"
+	emissionsModuleTypes "github.com/zeta-chain/zetacore/x/emissions/types"
+	fungibleModuleTypes "github.com/zeta-chain/zetacore/x/fungible/types"
+	observertypes "github.com/zeta-chain/zetacore/x/observer/types"
+)
+
+const MaxItemsForList = 10
+
+var Copy = map[string]bool{
+	slashingtypes.ModuleName:        true,
+	govtypes.ModuleName:             true,
+	crisistypes.ModuleName:          true,
+	feemarkettypes.ModuleName:       true,
+	paramstypes.ModuleName:          true,
+	upgradetypes.ModuleName:         true,
+	evidencetypes.ModuleName:        true,
+	vestingtypes.ModuleName:         true,
+	fungibleModuleTypes.ModuleName:  true,
+	emissionsModuleTypes.ModuleName: true,
+	authz.ModuleName:                true,
+}
+var Skip = map[string]bool{
+	evmtypes.ModuleName:          true,
+	stakingtypes.ModuleName:      true,
+	genutiltypes.ModuleName:      true,
+	authtypes.ModuleName:         true,
+	banktypes.ModuleName:         true,
+	distributiontypes.ModuleName: true,
+	group.ModuleName:             true,
+}
+
+var Modify = map[string]bool{
+	crosschaintypes.ModuleName: true,
+	observertypes.ModuleName:   true,
+}
+
+func CmdParseGenesisFile() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "parse-genesis-file [import-genesis-file] [optional-genesis-file]",
+		Short: "Parse the provided genesis file and import the required data into the optionally provided genesis file",
+		Args:  cobra.MaximumNArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx := client.GetClientContextFromCmd(cmd)
+			cdc := clientCtx.Codec
+			genesisFilePath := filepath.Join(app.DefaultNodeHome, "config", "genesis.json")
+			if len(args) == 2 {
+				genesisFilePath = args[1]
+			}
+			genDoc, err := GetGenDoc(genesisFilePath)
+			importData, err := GetGenDoc(args[0])
+
+			err = ImportDataIntoFile(genDoc, importData, cdc)
+			if err != nil {
+				return err
+			}
+
+			err = genutil.ExportGenesisFile(genDoc, genesisFilePath)
+			if err != nil {
+				return err
+			}
+
+			return nil
+		},
+	}
+	return cmd
+}
+
+func ImportDataIntoFile(genDoc *types.GenesisDoc, importFile *types.GenesisDoc, cdc codec.Codec) error {
+
+	appState, err := genutiltypes.GenesisStateFromGenDoc(*genDoc)
+	if err != nil {
+		return err
+	}
+	importAppState, err := genutiltypes.GenesisStateFromGenDoc(*importFile)
+	if err != nil {
+		return err
+	}
+	moduleList := app.InitGenesisModuleList()
+	for _, m := range moduleList {
+		if Skip[m] {
```

Comment:
I wanted to keep it for the following two very minor reasons 
- Adding the skip, makes sure we don't do the two other checks when not needed 
- It is slightly more readable for someone unfamiliar with Go, as the names Copy, Modify, and Skip are easy to understand. This would mostly be used by non protocol team members